session/session.go

```diff
@@ -1870,6 +1882,9 @@ func (s *session) ExecutePreparedStmt(ctx context.Context, stmtID uint32, args [
 	if !ok {
 		return nil, errors.Errorf("invalid CachedPrepareStmt type")
 	}
+	if variable.TopSQLEnabled() && preparedStmt.SQLDigest != nil {
+		ctx = topsql.SetSQLLabels(ctx, preparedStmt.NormalizedSQL, preparedStmt.SQLDigest.String())
```

Comment:
I think it will be good to simply separate into two functions:

```golang
topsql.LinkSQLTextWithDigest(...)
ctx = tracecpu.WrapCtxWithSQLLabels(ctx, digest)
```

It is simply easier to be understand, even using longer code.